src/aliceVision/sfm/pipeline/sequential/ReconstructionEngine_sequentialSfM.cpp

```diff
@@ -1415,59 +1430,281 @@ bool ReconstructionEngine_sequentialSfM::Resection(const std::size_t viewIndex)
+      const IntrinsicBase* camJ = scene.GetIntrinsics().at(viewJ->getIntrinsicId()).get();
+      const Pose3 poseI = scene.getPose(*viewI);
+      const Pose3 poseJ = scene.getPose(*viewJ);
+      const Vec2 xI = _featuresPerView->getFeatures(I, track.descType)[track.featPerView.at(I)].coords().cast<double>();
+      const Vec2 xJ = _featuresPerView->getFeatures(J, track.descType)[track.featPerView.at(J)].coords().cast<double>();
+  
+      // -- Triangulate:
+      TriangulateDLT(camI->get_projective_equivalent(poseI), 
+                     camI->get_ud_pixel(xI), 
+                     camJ->get_projective_equivalent(poseJ), 
+                     camI->get_ud_pixel(xJ), 
+                     &X_euclidean);
+      
+      // -- Check:
+      //  - angle (small angle leads imprecise triangulation)
+      //  - positive depth
+      //  - residual values
+      // TODO assert(acThresholdIt != _map_ACThreshold.end());
+      const auto& acThresholdItI = _map_ACThreshold.find(I);
+      const auto& acThresholdItJ = _map_ACThreshold.find(J);
+      const double& acThresholdI = (acThresholdItI != _map_ACThreshold.end()) ? acThresholdItI->second : 4.0;
+      const double& acThresholdJ = (acThresholdItJ != _map_ACThreshold.end()) ? acThresholdItJ->second : 4.0;
+      
+      if (AngleBetweenRay(poseI, camI, poseJ, camJ, xI, xJ) < _minAngleForTriangulation || 
+          poseI.depth(X_euclidean) < 0 || 
+          poseJ.depth(X_euclidean) < 0 || 
+          camI->residual(poseI, X_euclidean, xI).norm() > acThresholdI || 
+          camJ->residual(poseJ, X_euclidean, xJ).norm() > acThresholdJ)
+        continue;
+    }
+    else 
+    {
+      /* -------------------------------------------------------
+       *    N obsevations (N>2) : triangulation using LORANSAC 
+       * ------------------------------------------------------- */ 
+     
+      // -- Prepare:
+      Mat2X features(2, observations.size()); // undistorted 2D features (one per pose)
+      std::vector< Mat34 > Ps; // projective matrices (one per pose)
+      {
+        const track::Track & track = _map_tracks.at(trackId);
+        
+        int i = 0;
+        for (const IndexT & viewId : observations)
+        {
+          const View* view = scene.GetViews().at(viewId).get();
+          const IntrinsicBase* cam = scene.GetIntrinsics().at(view->getIntrinsicId()).get();
+          const Vec2 x_ud = cam->get_ud_pixel(_featuresPerView->getFeatures(viewId, track.descType)[track.featPerView.at(viewId)].coords().cast<double>()); // undistorted 2D point
+          features(0,i) = x_ud(0); 
+          features(1,i) = x_ud(1);  
+          Ps.push_back(cam->get_projective_equivalent(scene.getPose(*view)));
+          i++;
+        }
+      }
+      
+      // -- Triangulate: 
+      Vec4 X_homogeneous = Vec4::Zero();
+      std::vector<std::size_t> inliersIndex;
+      
+      TriangulateNViewLORANSAC(features, Ps, &X_homogeneous, &inliersIndex);
```

Comment:
The last parameter `threshold` is not set and the default value is `4.0` which is different from `_minAngleForTriangulation=3.0` used in DLT.

Reply:
Ah no, it's not related.